pyqir-generator/pyqir/generator/_native.pyi

```diff
@@ -58,13 +59,104 @@ def const(ty: Type, value: Union[int, float]) -> Value:
 class Builder:
     """An instruction builder."""
 
+    def and_(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical and instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def or_(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical or instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def xor(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts a bitwise logical exclusive or instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The result.
+        """
+        ...
+
+    def add(self, lhs: Value, rhs: Value) -> Value:
+        """
+        Inserts an addition instruction.
+
+        :param lhs: The left-hand side.
+        :param rhs: The right-hand side.
+        :returns: The sum.
+        """
+        ...
+
+    def sub(self, lhs: Value, rhs: Value) -> Value:
```

Comment:
`builder.neg(x)` is not supported directly, but is equivalent to `builder.sub(const(your_int_type, 0), x)`.